app/scenes/channel/channel_drawer_button.js

```diff
@@ -16,6 +17,9 @@ import {getTheme} from 'service/selectors/entities/preferences';
 function ChannelDrawerButton(props) {
     let badge;
     let badgeCount = props.mentionCount;
+
+    PushNotification.setApplicationIconBadgeNumber(props.mentionCount || 0);
```

Comment:
Could this be moved to be with the rest of the notification handling code? I feel like this code is going to be hard to find here if we ever go looking for it